lib/mixins/navigate.js

```
@@ -211,6 +212,9 @@ async function navToUrl (url) {
         `Could not enable console events on the page '${this.pageIdKey}' (app '${this.appIdKey}') ` +
         `within ${CONSOLE_ENABLEMENT_TIMEOUT_MS}ms. This might be an Inspector bug.`
       );
+      throw new errors.TimeoutError(`Could not enable console events after the page load within ` +
+        `${CONSOLE_ENABLEMENT_TIMEOUT_MS}ms. The Web Inspector/Safari may need to be restarted. ` +
+        `Original error: ${err.message}`);
```

the original error would look like `timeout` (actually this is the message that we can pass as an argument to the bluebird's `timeout` API), so there is no need to print it here

---

Also I am still not 100% convinced we should throw an error if we have actually failed to wait until the page load is completed. Maybe there is some really slow page...
Perhaps in such case it would even make sense to not wait for a response from this RPC call at all...

---

yea, I don't have much slow page like over 10 sec, so I haven't tested well...


At least no response by Web Inspector could occur kind of frequently in Safari 17. When I repeated a URL to repeat navigating 10 times, like this no response case occurred. Based on the Safari's GUI, the progress bar was completed in this initial no response case.

Once this no response case occurred, Safari's GUI no longer show progress bar/load a URL. `driver.title` also had no response so long (at least over a minute, I killed my local Appium server then). So, checking page update with `driver.title` etc does not work well as no response under this condition right now, so I thought raising error was better than keeping no response under case so that making test scenario fail fast

---

Let me merge it as is and then I will propose a solution for the "not loaded" case in a different PR

---

isn't there already a user specified page load timeout we could use for page load specifically?

---

yes, it is `this.pageLoadMs` and we use it. Check my next change at https://github.com/appium/appium-remote-debugger/pull/360

---

ah got it 👍